src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -65,57 +66,56 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
             if (actionDescriptor.BoundProperties.Count == 0 &&
                 actionDescriptor.Parameters.Count == 0)
             {
-                return Task.FromResult<IDictionary<string, object>>(
-                    new Dictionary<string, object>(StringComparer.Ordinal));
-            }
-            else if (actionDescriptor.BoundProperties.Count == 0)
-            {
-                return BindActionArgumentsCoreAsync(controllerContext, actionDescriptor);
-            }
-            else
-            {
-                return BindActionArgumentsAndPropertiesCoreAsync(
-                    controllerContext,
-                    controller,
-                    actionDescriptor);
+                return TaskCache.CompletedTask;
             }
-        }
 
-        private async Task<IDictionary<string, object>> BindActionArgumentsCoreAsync(
-            ControllerContext controllerContext,
-            ControllerActionDescriptor actionDescriptor)
-        {
-            var valueProvider = await CompositeValueProvider.CreateAsync(
-                controllerContext,
-                controllerContext.ValueProviderFactories);
-
-            var actionArguments = await PopulateArgumentsAsync(
-                controllerContext,
-                actionDescriptor.Parameters,
-                valueProvider);
-            return actionArguments;
+            return BindArgumentsCoreAsync(controllerContext, controller, arguments);
```

Comment:
A lot of this could happen w/o changing any other files. Seems like there's three overlapping things in this PR:

1. Remove duplication and strangeness inside `ControllerArgumentBinder`. All `private`, all good.
2. Avoiding a `Task<T>` return value in `IControllerActionArgumentBinder`. Probably enough goodness there to take the `public` API change hit.
3. Changing `IControllerActionArgumentBinder` name to match filename and name of this default implementation. I'm less sure about this part's worth.